File: API-Rockstars/API-Rockstars/Models/Rockstar.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Reflection.Metadata;
using Newtonsoft.Json.Converters;

namespace API_Rockstars.Models
{
    public class Rockstar
    {
        [Key] public Guid Id { get; set; }
        [Required] public string Name { get; set; }
        [Column(TypeName = "varchar(MAX)")] public string? Image { get; set; }
        public string? Description { get; set; }
        [NotMapped] public string? Role { get; set; }
        public string? LinkedIn { get; set; }
        public string? Twitter { get; set; }
        [EmailAddress] public string? Email { get; set; }
        [Phone] public string? Phone { get; set; }
    }
}

File: kdp-frontend/src/models/Rockstar.ts
interface SocialMediaTypes {
  facebookLink: string;
  linkedInLink: string;
  twitterLink: string;
}

export interface BaseRockstarShape {
  id: string;
  name: string;
}

export class RockstarShape implements BaseRockstarShape{
  id: string;
  name: string;
  image: string;
  description: string;
  role: string;
  linkedIn?:string;
  twitter?:string;
  email:string;
  phone?:string;
  rockstarSocial?: SocialMediaTypes;
  constructor(id: string, name: string, image: string, description: string, role: string, email:string, rockstarSocial?:SocialMediaTypes) {
    this.id = id;
    this.name = name;
    this.image = image;
    this.description = description;
    this.role = role;
    this.email = email;
    this.rockstarSocial = rockstarSocial;
  }
}
